packages/client/src/grant-token.ts

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import { z } from 'zod';
+
+const TokenSetParametersSchema = z.object({
+  access_token: z.string(),
+  expires_in: z.number(),
+  id_token: z.string(),
+  refresh_token: z.string(),
+});
+
+export type TokenSetParameters = z.infer<typeof TokenSetParametersSchema>;
+
+export const grantTokenByAuthorizationCode = async (
+  endpoint: string,
+  code: string,
+  redirectUri: string,
+  codeVerifier: string
+): Promise<TokenSetParameters> => {
+  const parameters = new URLSearchParams();
+  parameters.append('grant_type', 'authorization_code');
+  parameters.append('code', code);
+  parameters.append('redirect_uri', redirectUri);
+  parameters.append('code_verifier', codeVerifier);
+
+  const { data } = await axios.post<TokenSetParameters>(endpoint, parameters, {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+  });
+
+  return TokenSetParametersSchema.parse(data);
```

Comment:
in general, we should assume api returns the correct schema, otherwise the generic param in `axios.post<>` is meaningless.

Reply:
discussed offline, no harm for checking this schema, we can also prevent some incidents like mistyped URL. i'm good with this. cc @simeng-li